src/Entity/Donation.php

```diff
@@ -242,4 +258,32 @@ public function getRetryPayload(): array
 
         return $payload;
     }
+
+    public function getSubscriptionEndedAt(): ?\DateTime
+    {
+        return $this->subscriptionEndedAt;
+    }
+
+    public function setSubscriptionEndedAt(?\DateTime $subscriptionEndedAt): void
+    {
+        $this->subscriptionEndedAt = $subscriptionEndedAt;
+    }
+
+    public function nextDonationAt(): \DateTime
+    {
+        if (!$this->hasSubscription()) {
+            throw new \LogicException('Donation without subscription can\'t have next donation date.');
+        }
+
+        $day = $this->donatedAt->format('d');
+        $dayNow = date('d');
+        $strDay = $this->donatedAt->format('Y/m/_ h:i:s');
+        $strDay = str_replace('_', $day, $strDay);
+        $dateTime = \DateTime::createFromFormat('Y/m/d h:i:s', $strDay);
```

Comment:
what?
```
$day = $this->donatedAt->format('d');
$strDay = $this->donatedAt->format('Y/m/_ h:i:s');
$strDay = str_replace('_', $day, $strDay);
$dateTime = \DateTime::createFromFormat('Y/m/d h:i:s', $strDay);
```
===
```
$dateTime = $this->donatedAt;
```

Reply:
Sorry, I correct it and create a test for this method